### PR TITLE
Clean up generics

### PR DESCRIPTION
**Pull request by nstepien**

### CODE REVIEW

File: src/Cell.tsx [hunk 1]
```diff
@@ -127,6 +127,6 @@ function Cell<R, SR>(
   );
 }
 
-export default memo(forwardRef(Cell)) as <R, SR = unknown>(
+export default memo(forwardRef(Cell)) as <R, SR>(
```
Comment (nstepien) [inline src/Cell.tsx:130]: I don't think we need a default here: we need to pass a `column` prop, which will allow TS to infer the `SR` from it, so it should be fine.

image

This also applies to other components in this PR like `Row` and `TextEditor`.

File: src/types.ts [hunk 2]
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (nstepien) [inline src/types.ts:1]: 👍

File: src/types.ts [hunk 3]
```diff
@@ -74,15 +73,15 @@ export interface Position {
   rowIdx: number;
 }
 
-export interface FormatterProps<TRow = any, TSummaryRow = any> {
+export interface FormatterProps<TRow, TSummaryRow = unknown> {
```
Comment (nstepien) [inline src/types.ts:76]: I think it's fair to at least ask for `TRow`. Users can still use `<unknown>` or `<any>` if they really want to.

**Review by amanmahajan7 — APPROVED** [review]
Nice 👍

**Pull request merged**